implementation/discovery/api/src/main/java/org/inaetics/dronessimulator/discovery/api/Discoverer.java

```diff
@@ -28,6 +27,11 @@
     void unregister(Instance instance) throws IOException;
 
     /**
+     * Updates the properties associated with an instance. The instance must exist!
+     */
+     Instance updateProperties(Instance instance, Map<String, String> properties) throws IOException;
```

Comment:
Is needed for architecture management. The current lifecycle state is published with discovery to make use of the persistent storage. However, this state changes and so the properties have to be updated